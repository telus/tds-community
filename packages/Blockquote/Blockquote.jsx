import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Heading from '@tds/core-heading'
import Box from '@tds/core-box'
import { safeRest } from '@tds/util-helpers'
import { media } from '@tds/core-responsive'
import ColoredTextProvider from '../../shared/components/ColoredTextProvider/ColoredTextProvider'

const Wrapper = styled.blockquote({
  borderLeft: '4px solid purple',
  ...media.until('md').css({
    paddingLeft: '2rem',
  }),
  ...media.from('md').css({
    paddingLeft: '3rem',
  }),
})

/**
 * The Blockquote component is used to highlight text as a `blockquote` html tag.
 *
 * @version ./package.json
 */
const Blockquote = ({ children, ...rest }) => (
  <Wrapper {...safeRest(rest)}>
    <Box vertical={4}>
      <Heading level="h2" tag="span">
        <ColoredTextProvider>{children}</ColoredTextProvider>
      </Heading>
    </Box>
  </Wrapper>
)

Blockquote.propTypes = {
  /**
   * Specifies the text to be displayed within Blockquote.
   */
  children: PropTypes.node.isRequired,
}

export default Blockquote
