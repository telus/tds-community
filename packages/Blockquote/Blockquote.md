The `Blockquote` component wraps content in an HTML `<blockquote>` and `<span>` element and renders it as an `@tds/core-heading <h2>` component.

### Usage Criteria

- Use to highlight text to the user
- If using as a quote, always attribute a name to your quote
- Always include the correct quotation marks for EN (“”) and FR («»)
- If you’re quoting from an external source, include the source as a hyperlink
- If using a link within `Blockquote`, use `Link` from `@tds/core-link`
- Indicate cuts to quotes with ellipses (...) and edits with square brackets `[`...`]`

```jsx
<Blockquote>
  “Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
  labore et dolore magna aliqua” - <TdsLink href="https://tds.telus.com/">Name</TdsLink>
</Blockquote>
```
