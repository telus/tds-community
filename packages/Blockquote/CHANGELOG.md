# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.1.0](https://github.com/telus/tds-community/compare/@tds/community-blockquote@1.0.4...@tds/community-blockquote@1.1.0) (2021-09-21)


### Features

* **community-blockquote:** react 17 and styled components 5 ([ff4779e](https://github.com/telus/tds-community/commit/ff4779eb37924707bb35c47f865b7f4d35a71488))





## [1.0.4](https://github.com/telus/tds-community/compare/@tds/community-blockquote@1.0.3...@tds/community-blockquote@1.0.4) (2020-09-09)


### Bug Fixes

* **community-blockquote:** update util-helpers version ([eb29f6a](https://github.com/telus/tds-community/commit/eb29f6add6f6095384e92f7c9ae315e42d257c3f))





## [1.0.3](https://github.com/telus/tds-community/compare/@tds/community-blockquote@1.0.2...@tds/community-blockquote@1.0.3) (2020-04-13)

**Note:** Version bump only for package @tds/community-blockquote





## [1.0.2](https://github.com/telus/tds-community/compare/@tds/community-blockquote@1.0.1...@tds/community-blockquote@1.0.2) (2020-03-19)

**Note:** Version bump only for package @tds/community-blockquote





## [1.0.1](https://github.com/telus/tds-community/compare/@tds/community-blockquote@1.0.0...@tds/community-blockquote@1.0.1) (2019-09-17)

**Note:** Version bump only for package @tds/community-blockquote





# 1.0.0 (2019-06-13)

**Note:** Version bump only for package @tds/community-blockquote
