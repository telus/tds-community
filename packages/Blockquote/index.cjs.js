const Blockquote = require('./dist/index.cjs')

module.exports = Blockquote
