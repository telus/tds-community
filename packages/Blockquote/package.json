{
  "name": "@tds/community-blockquote",
  "version": "1.0.0-0",
  "description": "",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telus/tds-community.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telus/tds-community/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": "^16.8.2",
    "react-dom": "^16.8.2",
    "styled-components": "^4.1.3"
  },
  "dependencies": {
    "prop-types": "^15.6.2",
    "@tds/core-heading": "^1.1.5",
    "@tds/core-box": "^1.0.1",
    "@tds/core-responsive": "^1.1.0"
  },
  "devDependencies": {
    "@tds/shared-safe-rest": "^1.0.0"
  }
}
