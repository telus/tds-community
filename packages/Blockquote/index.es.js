import Blockquote from './dist/index.es'

export default Blockquote
