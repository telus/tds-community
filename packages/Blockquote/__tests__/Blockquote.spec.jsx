import React from 'react'
import { mount, shallow } from 'enzyme'

import Blockquote from '../Blockquote'

describe('Blockquote', () => {
  const defaultProps = {
    children: 'This is an example of Blockquote',
  }

  const doMount = (props = {}) => mount(<Blockquote {...defaultProps} {...props} />)

  const doShallow = (props = {}) => shallow(<Blockquote {...defaultProps} {...props} />)

  it('renders', () => {
    const blockQuote = doMount()

    expect(blockQuote).toMatchSnapshot()
  })

  it('does other things', () => {
    const blockQuote = doMount()

    expect(blockQuote).toExist()
  })

  it('passes additional attributes to the element', () => {
    const blockQuote = doMount({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(blockQuote).toHaveProp('id', 'the-id')
    expect(blockQuote).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const blockQuote = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(blockQuote).not.toHaveProp('className', 'my-custom-class')
    expect(blockQuote).not.toHaveProp('style')
  })
})
