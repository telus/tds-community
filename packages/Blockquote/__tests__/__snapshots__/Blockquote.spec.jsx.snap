// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote renders 1`] = `
.c0 {
  border-left: 4px solid purple;
}

@media (max-width:768px) {
  .c0 {
    padding-left: 2rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding-left: 3rem;
  }
}

<Blockquote>
  <Blockquote__Wrapper>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Blockquote__Wrapper-gnvw5h-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "border-left: 4px solid purple; @media (max-width: 768px) {
  padding-left: 2rem;
} @media (min-width: 768px) {
  padding-left: 3rem;
}",
            ],
          },
          "displayName": "Blockquote__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Blockquote__Wrapper-gnvw5h-0",
          "target": "blockquote",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <blockquote
        className="c0"
      >
        <Box
          inline={false}
          tag="div"
          vertical={4}
        >
          <div
            className="TDS_Box-modules__verticalPadding-4___QMNNa"
          >
            <Heading
              invert={false}
              level="h2"
              tag="span"
            >
              <span
                className="TDS_Heading-modules__h2___l3LmI TDS_Heading-modules__wordBreak___UVmND TDS_Spacing-modules__noSpacing___XPYDG TDS_Heading-modules__secondary___3skcz"
              >
                <ColoredTextProvider
                  className={null}
                >
                  <div>
                    This is an example of Blockquote
                  </div>
                </ColoredTextProvider>
              </span>
            </Heading>
          </div>
        </Box>
      </blockquote>
    </StyledComponent>
  </Blockquote__Wrapper>
</Blockquote>
`;
